src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
+	
+	public AbstractDAO(BeanSession session) {
```

Comment:
You're right.. I just don't like the idea of overriding the constructor on each subclass, but I can live with that..